packages/babel-helpers/src/helpers.js

```diff
@@ -860,10 +860,24 @@ helpers.arrayWithHoles = helper("7.0.0-beta.0")`
 
 helpers.iterableToArray = helper("7.0.0-beta.0")`
   export default function _iterableToArray(iter) {
-    if (
-      Symbol.iterator in Object(iter) ||
-      Object.prototype.toString.call(iter) === "[object Arguments]"
-    ) return Array.from(iter);
+    const prototypeWhiteList = [
+      '[object Arguments]',
+    ];
+    let hasSymbol = false;
+    try {
```

Comment:
Why is a try/catch needed here instead of just checking if Symbol.iterator exists or not?

Reply:
You'd have to check if `Symbol` exists before checking if `Symbol.iterator` exists. I squashed the previous commit, but checking if Symbol exists would bloat the code up, as explained the one of the previous comments.

Doing it this way removes the need for `typeof` checks and lets the browser pass the Symbol check failure to `catch`. Since we're not actually using `Symbol.iterator` the result would be same. If it can't check the object to see if it has it, then it wouldn't have it anyway.

Reply:
try/catch has a performance penalty that's far worse than a few extra gzippable bytes.

Reply:
As I mentioned in the other comment, this is the third try/catch in the same helper file. I don't see why this specific one should be treated any differently.

Reply:
because in that case, it's being used to catch errors thrown by user code (the iterator method). In this case it's used to catch errors in your code itself (attempting unchecked member access on a maybe-not-object).

Reply:
Ah, in the `next()` function. Okay, back to the drawing board.